### PR TITLE
[fix](function) error scale set in unix_timestamp (#36110)

### PR DESCRIPTION
**Pull request by Mryange**
## Proposed changes
https://github.com/apache/doris/pull/36110
```
mysql [test]>set DEBUG_SKIP_FOLD_CONSTANT = true;
Query OK, 0 rows affected (0.00 sec)

mysql [test]>select cast(unix_timestamp("2024-01-01",'yyyy-MM-dd') as bigint);
+------------------------------------------------------------+
| cast(unix_timestamp('2024-01-01', 'yyyy-MM-dd') as BIGINT) |
+------------------------------------------------------------+
|                                           1704038400000000 |
+------------------------------------------------------------+
```
now
```
mysql [test]>select cast(unix_timestamp("2024-01-01",'yyyy-MM-dd') as bigint);
+------------------------------------------------------------+
| cast(unix_timestamp('2024-01-01', 'yyyy-MM-dd') as BIGINT) |
+------------------------------------------------------------+
|                                                 1704038400 |
+------------------------------------------------------------+
1 row in set (0.01 sec)
```

The column does not have a scale set, but the cast uses the scale to perform the cast.

## Proposed changes

Issue Number: close #xxx

### CODE REVIEW

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

File: regression-test/suites/query_p0/sql_functions/datetime_functions/test_date_function.groovy [hunk 1]
```diff
@@ -441,13 +441,15 @@ suite("test_date_function") {
     // UNIX_TIMESTAMP
     def unin_timestamp_str = """ select unix_timestamp() """
     assertTrue(unin_timestamp_str[0].size() == 1)
+
```
Comment (zhiqiang-hhhh) [inline regression-test/suites/query_p0/sql_functions/datetime_functions/test_date_function.groovy:444]: sql """set DEBUG_SKIP_FOLD_CONSTANT = true;""" is missing?

**Comment by zhiqiang-hhhh** [comment]
Is this picked from https://github.com/apache/doris/pull/36110 ? If so, why `sql """set DEBUG_SKIP_FOLD_CONSTANT = true;"""` is missing?

**Comment by Mryange** [comment]
> Is this picked from #36110 ? If so, why `sql """set DEBUG_SKIP_FOLD_CONSTANT = true;"""` is missing?

Because no such variable in 2.1

java.sql.SQLException: errCode = 2, detailMessage = Unknown system variable 'DEBUG_SKIP_FOLD_CONSTANT',the similar variables are {'forbid_unknown_col_stats', 'enable_strong_consistency_read', 'enable_function_pushdown'}

**Pull request merged**